asmcomp/asmlink.ml

```diff
@@ -276,13 +288,15 @@ let call_linker_shared file_list output_name =
 
 let link_shared ~ppf_dump objfiles output_name =
   Profile.record_call output_name (fun () ->
-    let units_tolink = List.fold_right scan_file objfiles [] in
+    let units_tolink, objfiles_rev =
+      List.fold_right scan_file objfiles ([], [])
```

Comment:
I'm confused about the "rev" part, here and below.  This is a fold right, not a fold left, so it doesn't magically reverse lists.  E.g.
```
  List.fold_right (fun a l -> a :: l) l []
```
is the identity, not list reverse.

Reply:
Oh man, that's embarrassing 🙁